src/Agent.cs

```diff
@@ -62,7 +62,8 @@ public Agent(String apiKey)
       _collector = new Collector(apiKey);
 
       var validationOptions = RegexOptions.IgnoreCase | RegexOptions.Compiled | RegexOptions.ExplicitCapture;
-      _validateMetric = new Regex(@"^([\d\w\-_]+\.)*[\d\w\-_]+$", validationOptions);
+      _validateMetric = new Regex(@"^([a-zA-Z0-9\-_]+\.)*[a-zA-Z0-9\-_]+$", validationOptions);
```

Comment:
I do not know off the top of my head - we could use explicit single line/multi line.  FWIW, our one bad character tests uses \n as the bad character.

I'll add tests with some -_0-9 in them.